### ISSUE TITLE
/usr/include/wchar.h:39:11: fatal error: 'stdarg.h' file not found

### ISSUE THREAD

**Issue by timtro** [lvzixun/clang-complete#37]
Thanks in advance. The plugin is largely working for me and seems amazing.

I'm running arch linux. I am using clang++ to compile my code without problems, so my clang implementation doesn't seem to be the issue. I'm guessing there is some sort of search path problem or a link problem.

To install, I modified the makefile by commenting out the linux_config: lines:

```
linux_config:
    #sudo apt-get install clang
    #sudo ln -sf /usr/lib/llvm-3.5/lib/libclang-3.5.so.1 ./lib/libclang.so
```

and changed the ST3 variable accordingly:

```
# linux
ifeq ($(UNAME_S), Linux)
    CC = gcc
    CFLAGS = -g -Wall -fPIC
    ST3 = ~/.config/sublime-text-3/Packages/Clang-Complete
    LIBCC = lib/libcc.so
endif
```

I ran `make` (not `make linux`) by accident, and then ran `make linux`, but everything seemed otherwise fine.

Whenever I save a document however, what I assume is the static analyser output appears on the bottom of the sublime window:

```
/usr/include/wchar.h:39:11: fatal error: 'stdarg.h' file not found
```

Searching the error bring up nothing recent. The most relevant thing I can find is a bug report from 2013 and the problem was a faulty symbolic link in Ubuntu... so, nothing relevant really. And I don't know enough about anything to begin troubleshooting.

Help would be MUCH appreciated.

**Comment by druppy** [comment]
This sounds to me like you need to apply some system path's in you settings, as it looks like the plugin cant find the header file.

Look inside the `cc.sublime-setting` file, there you will find templates for both option settings and include_options.

On my debian jessie, the settings need this :

``` json
"include_options": [
        "-isystem", "/usr/include",
        "-isystem", "/usr/include/c++/*",
        "-isystem", "/usr/lib/clang/3.7/include",
]
```

This works quite well for me, hope you end with the same success :-)

**Comment by timtro** [comment]
Thanks, druppy. I had done that, and embarrassingly enough had a typo. I corrected it and fixed the problem.

I was trigger happy on this issue, after having tried another Clang completer and had to write out its use of the `-cc1` flag because that option apparently has a bug (that is supposedly fixed) that causes an incomplete search path. I had to replace it with `-fsyntax-only -Xclang`

A shot of encouragement: I've tried at least three clang based Sublime Text plugins, and as far as I'm concerned, this is the only one that works well.
